CY-3564 - Improve widgets documentation (#1440)
cloudify-cosmo/docs.getcloudify.org: content/working_with/console/widgets/topology.md

### Presentation

Each of the blueprint's nodes is displayed as a square container that can contain other nodes. 
Each node has a name, and an icon (upper right corner) to indicate its [node type]({{< relref "developer/blueprints/spec-node-types.md" >}}). 

[Relationships]({{< relref "developer/blueprints/spec-relationships.md" >}}) between nodes are indicated with arrows that start at the connected node and end at the target node.

The number of node instances is marked in a bullet beside the node's type icon.

> Each node is provided with [plugin]({{< relref "developer/blueprints/spec-plugins.md" >}}) icon (top left corner) which displays the plugin from which the node comes from.

Reviewer: Not every node type comes from a plugin - maybe worth to mention the generic "plug" icon that is used for built-in types?

Author: Yep, added more detailed description.

For **Component** nodes you can also see bottom right corner icons showing all plugins used by the component's internal nodes. 

![Topology - multi plugins]( /images/ui/widgets/topology-widget_multi-plugins.png )

Node types used in [service composition]({{< relref "working_with/service_composition/_index.md" >}}) are displayed in special way which allows user to expand it (display internal nodes) or drill-down to deployment (opens node's deployment page).

![Topology - service composition]( /images/ui/widgets/topology-widget_component-node.png )

### Actions

In Topology widget you can:
 

> * panning around the view (drag'n'drop outside node)

Reviewer: ```suggestion
* pan around the view (drag'n'drop outside node)
```

* change location of the nodes (drag'n'drop inside node)
* zoom in/out (using mouse wheel)     

There is also a toolbar in the right upper corner:

![Topology - toolbar]( /images/ui/widgets/topology-widget_toolbar.png )

It allows you to make changes in the view:

* **Zoom in** the topology
* **Zoom out** the topology
* **Fit topology to screen**
* **Save layout** - save location of the nodes (the location is saved per blueprint per user) 
* **Revert layout changes** - revert the location of the nodes to the previous state

> * **Auto layout** - automatically calculate the best position for all the nodes

Reviewer: I doubt auto layout really calculates "best" position :)

Author: I was fighting with myself when writing it :-)

### Terraform support 

> The Terraform nodes created using [Terraform plugin]({{< relref "working_with/official_plugins/orchestration/terraform.md" >}})

Reviewer: ```suggestion
Terraform nodes created using [Terraform plugin]({{< relref "working_with/official_plugins/orchestration/terraform.md" >}})
```

> are treated in a special way. There are action icons in the bottom left corner of such nodes:

Reviewer: ```suggestion
are treated in a special way. There are dedicated action icons in the bottom left corner of such nodes:
```

* **Info** - shows Terraform node details in tabular view

> * **Expand / Collapse** - similar functionality as for Component and SharedResources nodes, but for Terraform node

Reviewer: ```suggestion
* **Expand / Collapse** - similar functionality as for Component and SharedResource nodes, but for Terraform node
```

The number of node instances is marked in a bullet beside the node's type icon.

Each node is provided with an icon (top left corner). For built-in node types it is Cloudify logo. For node types coming from [plugins]({{< relref "developer/blueprints/spec-plugins.md" >}}) it is an icon selected during plugin upload (setting plugin icon is optional, by default a plug icon is used). 
See [Plugins widget]({{< relref "working_with/console/widgets/plugins.md" >}}) or [Plugins Catalog widget]({{< relref "working_with/console/widgets/pluginsCatalog.md" >}}) for more details.

For **Component** nodes you can also see bottom right corner icons showing all plugins used by the component's internal nodes. 

![Topology - multi plugins]( /images/ui/widgets/topology-widget_multi-plugins.png )

Node types used in [service composition]({{< relref "working_with/service_composition/_index.md" >}}) are displayed in special way which allows user to expand it (display internal nodes) or drill-down to deployment (opens node's deployment page).

![Topology - service composition]( /images/ui/widgets/topology-widget_component-node.png )

### Actions

In Topology widget you can:
 

> * pan around the view (drag'n'drop outside node)

Reviewer 2: ```suggestion
* Pan around the view (drag'n'drop outside node)
```

> * change location of the nodes (drag'n'drop inside node)

Reviewer 2: ```suggestion
* Change location of the nodes (drag'n'drop inside node)
```

> * zoom in/out (using mouse wheel)     

Reviewer 2: ```suggestion
* Zoom in/out (using mouse wheel)     
```

There is also a toolbar in the right upper corner:

![Topology - toolbar]( /images/ui/widgets/topology-widget_toolbar.png )

It allows you to make changes in the view:

* **Zoom in** the topology
* **Zoom out** the topology
* **Fit topology to screen**
* **Save layout** - save location of the nodes (the location is saved per blueprint per user) 
* **Revert layout changes** - revert the location of the nodes to the previous state
* **Auto layout** - automatically distribute the nodes on the canvas

### Terraform support 

Terraform nodes created using [Terraform plugin]({{< relref "working_with/official_plugins/orchestration/terraform.md" >}})
are treated in a special way. There are dedicated action icons in the bottom left corner of such nodes:

* **Info** - shows Terraform node details in tabular view

> * **Expand / Collapse** - similar functionality as for Component and SharedResource nodes, but for Terraform node

Reviewer 2: ```suggestion
* **Expand / Collapse** - opens/closes a view of the Terraform embedded topology, similar to the views presented for Component and SharedResource nodes.
```
